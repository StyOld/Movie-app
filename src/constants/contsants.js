export const UPDATE_AUTH = 'UPDATE_AUTH';
export const LOGOUT = 'LOGOUT';
export const TOGGLE_LOGINFORM = 'TOGGLE_LOGINFORM';
export const HIDE_LOGINFORM = 'HIDE_LOGINFORM';
export const GET_MOVIE_DETAILS = 'GET_MOVIE_DETAILS';
export const UPDATE_MOVIE_DETAILS = 'UPDATE_MOVIE_DETAILS';
export const FETCHING_MOVIES = 'FETCHING_MOVIES';
export const UPDATE_MOVIES = 'UPDATE_MOVIES';
export const ERROR_GET_MOVIES = 'ERROR_GET_MOVIES';
export const GET_BY_TYPE_MOVIES = 'GET_BY_TYPE_MOVIES';
export const GET_GENRES_LIST = 'GET_GENRES_LIST';
export const ERROR_GET_GENRES = 'ERROR_GET_GENRES';
export const TOGGLE_DROP_DOWN = 'TOGGLE_DROP_DOWN';
export const ERROR_DELETE_SESSION = 'ERROR_DELETE_SESSION';
export const CHANGE_PAGE = 'CHANGE_PAGE';
export const CLEAR_FILTERS = 'CLEAR_FILTERS';
export const CHANGE_FILTERS = 'CHANGE_FILTERS';
export const GET_BY_TYPE_MOVIE_DETAILS = 'GET_BY_TYPE_MOVIE_DETAILS';
export const ERROR_GET_BY_TYPE_MOVIE_DETAILS = 'ERROR_GET_BY_TYPE_MOVIE_DETAILS';