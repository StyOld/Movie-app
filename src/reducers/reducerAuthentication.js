import * as constants from '../constants/contsants';
import Cookies from 'universal-cookie';
const cookies = new Cookies();

export const initialState = {
    user: null,
    session_id: cookies.get('session_id'),
    isAuth: false,
    showModal: false,
    showDropDown: false
};

const reducerAuthentication = (state = initialState, action) => {
    switch(action.type) {
        case constants.UPDATE_AUTH:
            return {
                ...state,
                user: action.payload.user,
                session_id: action.payload.session_id,
                isAuth: true
            };

        case constants.LOGOUT:
            cookies.remove('session_id');
            return {
                ...state,
                session_id: null,
                user: null,
                isAuth: false
            };

        case constants.TOGGLE_LOGINFORM:
            return {
                ...state,
                showModal: !state.showModal
            };

        case constants.HIDE_LOGINFORM:
            return {
                ...state,
                showModal: false
            };

        case constants.TOGGLE_DROP_DOWN:
            return {
                ...state,
                showDropDown: !state.showDropDown
            };

        default:
            return state;
    }
};

export default reducerAuthentication;