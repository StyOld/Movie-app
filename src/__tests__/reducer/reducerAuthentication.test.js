import reducerAuthentication, {initialState} from '../../reducers/reducerAuthentication';
import * as constants from '../../constants/contsants';

describe('reducerAuthentication', () => {
    test('default', () => {
        expect(reducerAuthentication(undefined, {
            type: 'INIT'
        })
        ).toEqual(initialState)
    });

    test(`${constants.UPDATE_AUTH}`, () => {
        const action = {
            type: constants.UPDATE_AUTH,
            payload: {
                user: 'name',
                session_id: 1
            }
        };

        expect(reducerAuthentication(undefined, action)).toMatchSnapshot();
    });

    test(`${constants.LOGOUT}`, () => {
        const action = {
            type: constants.LOGOUT
            };

        const newState = reducerAuthentication(undefined, action);
        expect(newState.session_id).toBeNull();
        expect(newState.user).toBeNull();
        expect(newState.isAuth).toBeFalsy();
    });
});
