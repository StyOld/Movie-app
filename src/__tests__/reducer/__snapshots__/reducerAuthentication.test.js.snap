// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducerAuthentication UPDATE_AUTH 1`] = `
Object {
  "isAuth": true,
  "session_id": 1,
  "showDropDown": false,
  "showModal": false,
  "user": "name",
}
`;
